server/aws-lsp-codewhisperer/src/language-server/inline-completion/session/sessionManager.ts

```diff
@@ -287,17 +288,18 @@ export class SessionManager {
     /**
      * Singleton SessionManager class
      */
-    public static getInstance(): SessionManager {
-        if (!SessionManager._instance) {
-            SessionManager._instance = new SessionManager()
+    public static getInstance(type: 'COMPLETIONS' | 'EDITS' = 'COMPLETIONS'): SessionManager {
```

Comment:
It's to completely separate out the states of 2 functionalities. 1 for Completion and the other for Edits.